pymatgen/io/icet.py

```diff
@@ -0,0 +1,331 @@
+from __future__ import annotations
+
+import multiprocessing as multiproc
+import warnings
+from string import ascii_uppercase
+from time import time
+from typing import TYPE_CHECKING
+
+from monty.dev import requires
+
+from pymatgen.command_line.mcsqs_caller import Sqs
+from pymatgen.core import Structure
+from pymatgen.io.ase import AseAtomsAdaptor
+
+try:
+    from icet import ClusterSpace
+    from icet.tools import enumerate_structures
+    from icet.tools.structure_generation import _get_sqs_cluster_vector, _validate_concentrations, generate_sqs
+    from mchammer.calculators import compare_cluster_vectors
+except ImportError:
+    ClusterSpace = None
+
+
+if TYPE_CHECKING:
+    from typing import Any
+
+    from _icet import _ClusterSpace
+    from ase import Atoms
+
+
+@requires(ClusterSpace is not None, "IcetSQS requires the icet package. Use `pip install icet`")
```

Comment:
can we add a test for this error? would require `monkeypatching` the `ClusterSpace` variable in this module to `None`.

also would be good to change the error class from `Runtime` to `ImportError` once https://github.com/materialsvirtuallab/monty/pull/625 is merged

Reply:
I've added a test for the import but removed the `@requires` decorator, which I've found to be troublesome:
- It makes top-level functions no-longer top-level / non-pickle-able, which removes multiprocessing capability
- It is unaffected by monkeypatching. The following code, which should work with the `@requires` decorator, does not:
```
from pymatgen.io import icet as icet_interface
with pytest.MonkeyPatch.context() as monkeypatch:
    monkeypatch.setattr(icet_interface,"ClusterSpace",None)
    with pytest.raises(RuntimeError):
        icet_interface.IcetSQS(
        structure=self.stored_run["disordered_structure"],
        scaling=self.scaling,
        instances=None,
        cluster_cutoffs={2: 5.0}
    )
```
I consistently get `Failed: DID NOT RAISE <class 'RuntimeError'>`

Reply:
thanks @esoteric-ephemera! 

> which removes multiprocessing capability

interesting. @shyuep is this a known limitation? otherwise i'll open a monty issue to fix or document this.

either way thanks for fixing tests @esoteric-ephemera.


> you might want to wait to merge until [this icet issue](https://gitlab.com/materials-modeling/icet/-/issues/614) is resolved

is the draft status on this PR to wait for the above? otherwise lgtm